spec/enumerable_spec.rb

```diff
@@ -0,0 +1,203 @@
+# spec/enumerable_spec.rb
+
+require './enumerables.rb'
+
+RSpec.describe Enumerable do
+  let(:array) { [1, 2, 3, 4, 5] }
+  let(:arr_regex) { %w[dog door rod blade] }
+  let(:arr_nil) { [1, nil, false] }
+  let(:match_arr) { [1, 1, 1] }
+  let(:arr_empty) { [] }
+  let(:arr_count) { [1, 1, 1, 2, 3] }
+
+  describe '#my_each' do
+    it 'should return to Enumerator if no block given' do
+      expect(array.my_each).to be_a(Enumerator)
+      expect(array.my_each.to_a).to eq(array)
+    end
+
+    it 'should return same as each method if block given' do
+      expect(array.my_each { |elem| puts "return: #{elem}" }).to eq(array.each { |elem| puts "return: #{elem}" })
+    end
+
+    it 'should throw an error if given an argument' do
+      expect { array.my_each('argument') }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'should return to Enumerator if no block given' do
+      expect(array.my_each_with_index).to be_a(Enumerator)
+    end
+
+    it 'should return same as each_with_index method if block given' do
+      # rubocop:disable Layout/LineLength
+      expect(array.my_each_with_index { |elem, index| puts "#{elem} : #{index}" }).to eq(array.each_with_index { |elem, index| puts "#{elem} : #{index}" })
+      # rubocop:enable Layout/LineLength
+    end
+
+    it 'should throw an error if given an argument' do
+      expect { array.my_each_with_index('argument') }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#my_select' do
+    arr_even = [1, 2, 3, 8]
+    arr_odd = [6, 11, 13]
+    odd = [11, 13]
+    array = [3, 5, 'A', 'B']
```

Comment:
[OPTIONAL]

- [ ] you can also define the variables within your describe blocks using let

Reply:
Thank you for advice, just made the suggested changes.